pkg/atomicwriter/atomic_writer.go

```diff
@@ -0,0 +1,108 @@
+package atomicwriter
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/log/messages"
+)
+
+const deleteTempFileRetryInterval = time.Second * 5
+const deleteTempFileMaxRetries = 5
+
+// This package provides a simple atomic file writer which implements the
+// io.WriteCloser interface. This allows us to use AtomicWriter the way we
+// would use any other Writer, such as a Buffer. Additonally, this struct
+// takes the file path during construction, so the code which calls
+// `Write()` doesn't need to be concerned with the destination, just like
+// any other writer.
+type AtomicWriter struct {
+	path        string
+	permissions os.FileMode
+	tempFile    *os.File
+}
+
+func NewAtomicWriter(path string, permissions os.FileMode) (*AtomicWriter, error) {
```

Comment:
For the sake of limiting what is made public to that which is necessary I think we should make the struct definition private `AtomicWriter => atomicWriter` and `NewAtomicWriter` can return an interface`(io.WriteCloser, error)`.

Reply:
Changed in 3e1def1ede9eacd3d5d24d268036dd2770aaddaa